crates/core/component/auction/src/component/action_handler/dutch/schedule.rs

```diff
@@ -50,16 +79,17 @@ impl ActionHandler for ActionDutchAuctionSchedule {
             end_height
         );
 
-        // Check that the step count is positive.
+        // Check that the step count is at least 2. This is important
+        // because DA price interpolation assumes that `step_count-1` is positive.
         ensure!(
-            step_count > 0,
-            "step count MUST be greater than zero (got: {step_count})"
+            step_count >= 2,
+            "step count MUST be at least two (got: {step_count})"
         );
 
-        // Check that the step count is less than 1000.
+        // Check that the step count is less than 255.
         ensure!(
-            step_count <= 1000,
-            "the dutch auction step count MUST be less than 1000 (got: {step_count})",
+            step_count <= 255,
```

Comment:
this is related to the limit above, we don't gain much by having a larger step count capacity since the end game is to have a collection of DAs anyway, but this costs us precious precision space when we perform the price interpolation